index.js

```diff
@@ -12,11 +14,34 @@ exports.lintText = (str, opts) => {
 		delete opts.overrides;
 
 		const filename = path.relative(opts.cwd, opts.filename);
+
 		const foundOverrides = optionsManager.findApplicableOverrides(filename, overrides);
 		opts = optionsManager.mergeApplicableOverrides(opts, foundOverrides.applicable);
 	}
 
 	opts = optionsManager.buildConfig(opts);
+	const defaultIgnores = optionsManager.getIgnores({}).ignores;
+
+	if (opts.ignores && !arrayEqual(defaultIgnores, opts.ignores) && typeof opts.filename !== 'string') {
+		throw new Error(`xo: opts.filename must be string when providing opts.ignores. Received value "${opts.filename}" of type "${typeof opts.filename}" for opts.filename.`);
```

Comment:
This is a bit too verbose. People will most likely never hit this as we're the only ones using the API. I also don't see the point of the `xo:` prefix. It's clear from the stack trace where the error is coming from.

I'd prefer something like:

```
> The `ignores` option requires the `filename` and `cwd` option to be defined.
```